tests/mocks/featureTypesMocks.ts

```diff
@@ -0,0 +1,591 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+
+export const geoserverFeatureTypesListAllResponseMock = {
+  list: {
+    string: ['polygon_parts', 'best_feature', 'another_table'],
+  },
+};
+
+export const geoserverFeatureTypesListConfiguredResponseMock = {
+  featureTypes: {
+    featureType: [
+      {
+        name: 'best_feature',
+        href: 'http://localhost:8080/geoserver/rest/workspaces/test/datastores/test_polygon_parts/featuretypes/best_feature.json',
+      },
+      {
+        name: 'polygon_parts',
+        href: 'http://localhost:8080/geoserver/rest/workspaces/test/datastores/test_polygon_parts/featuretypes/polygon_parts.json',
+      },
+    ],
+  },
+};
+
+export const featureTypesListAllResponseMock = [
+  {
+    name: 'polygon_parts',
+  },
+  {
+    name: 'best_feature',
+  },
+  {
+    name: 'another_table',
+  },
+];
+
+export const featureTypesListConfiguredResponseMock = [
+  {
+    name: 'best_feature',
+    link: 'http://localhost:8080/geoserver/rest/workspaces/test/datastores/test_polygon_parts/featuretypes/best_feature.json',
+  },
+  {
+    name: 'polygon_parts',
+    link: 'http://localhost:8080/geoserver/rest/workspaces/test/datastores/test_polygon_parts/featuretypes/polygon_parts.json',
+  },
+];
+
+export const geoserverGetFeatureTypeResponseMock = {
+  featureType: {
+    name: 'bestFeature',
+    nativeName: 'bestStore',
+    namespace: {
+      name: 'test',
+      href: 'http://localhost:8080/geoserver/rest/namespaces/test.json',
+    },
+    title: 'polygon_parts',
+    keywords: {
+      string: ['features', 'polygon_parts'],
+    },
+    srs: 'EPSG:4326',
+    nativeBoundingBox: {
+      minx: -180,
+      maxx: 180,
+      miny: -90,
+      maxy: 90,
+      crs: 'EPSG:4326',
+    },
+    latLonBoundingBox: {
+      minx: -180,
+      maxx: 180,
+      miny: -90,
+      maxy: 90,
+      crs: 'EPSG:4326',
+    },
+    projectionPolicy: 'FORCE_DECLARED',
+    enabled: false,
+    store: {
+      '@class': 'dataStore',
+      name: 'test:bestStore',
+      href: 'http://localhost:8080/geoserver/rest/workspaces/test/datastores/bestStore.json',
+    },
+    serviceConfiguration: false,
+    internationalTitle: '',
+    internationalAbstract: '',
+    maxFeatures: 0,
+    numDecimals: 0,
+    padWithZeros: false,
+    forcedDecimal: false,
+    overridingServiceSRS: false,
+    skipNumberMatched: false,
+    circularArcPresent: false,
+    attributes: {
+      attribute: [
+        {
```

Comment:
check for required fields

Reply:
nullable are only: source_id, description. cities, countries